prettier.config.js

```diff
@@ -1,12 +1,4 @@
-// prettier.config.js or .prettierrc.js
 module.exports = {
     trailingComma: "all",
     tabWidth: 4,
-    semi: true,
-    singleQuote: true,
-    printWidth: 140,
```

Comment:
How do folks feel about `printWidth`? I generally find `80` to be harder to read and generally like `120` or so myself, but I'm flexible.

Reply:
I have an 80 column environment at my 9-5, so I'm pretty used to it.  `120` sounds reasonable to me though.  Just won't be able to use punch cards 😭

Reply:
I prefer 80 as an industry standard, but I don't think it's compatible with our current codebase... we have some really long lines that would be hard to read on 80 width but it would take a lot of work to find ALL of them and rewrite them into a tighter form factor (also you trade line width for total lines usually so it's a catch-22)

Reply:
The natural margin of code can see in front of me is about 100 chars or so.

Reply:
I'm going to change stance on this a bit.  [Prettier docs](https://prettier.io/docs/en/options.html#print-width) explicitly recommend against more than 80 chars.  The reasoning being that prettier will reformat short lines to be longer wherever possible, in contrast to ESLint's `max-len` which will only ever make lines shorter.

I opened a PR to compare 80 vs 120 printWidth: https://github.com/benjaminpjones/online-go.com/pull/1/files.  Beauty in the eye of the beholder and all that, but IMO too many lines are going into the gutter.  Therefore, if we can get closer to `80` and still be happy, I encourage that.

Reply:
It's a shame we can't turn that off and let eslint enforce with it's maxlength. I don't like particularly long lines, I like how they read at `80` columns *IF* the code is indented only maybe 0-4 times, but looking at the JSX when it's 15-20 levels deep, 80 column wrapping starts really hurting as it explodes everything vertically, and I breaking out content into multiple sections to reduce indent just turns that content into spaghetti.  I think what I really want is something like `60` column wrapping ignoring indentation columns or something along those lines, that doesn't seem to be a thing though.

Reply:
I hear that.  Each option definitely seems to have its downsides...

I guess where does that leave us? Should I drop the idea of integrating prettier? Or if prettier is still desirable, 80 or 120 or something in between?

> breaking out content into multiple sections to reduce indent just turns that content into spaghetti.

FWIW, I think highly indented code could benefit from being broken out (as long as function names make sense).  But that doesn't really help with existing code that would be "prettified"...

Reply:
I've started doing this in Play.tsx & User.tsx, where I'm messing around today.

Y'all  must detest my code, because I'm regularly out beyond column 100, even in logic.  With some refactoring this is down to 84:

```
    rengoReadyToStart = (challenge) => (
        challenge.rengo_black_team.length && challenge.rengo_white_team.length &&
        (challenge.rengo_black_team.length + challenge.rengo_white_team.length > 2)
```
        
I would really dislike having to put the onClick on the next line here:
```

                    <div className="rengo-challenge-buttons">
                        { (own_rengo_challenge || null) &&
                            <React.Fragment>
                                <div className='automatch-settings'>
                                    <button className='danger sm' onClick={this.cancelOwnChallenges.bind(self, this.state.rengo_list)}>
                                        {pgettext("Cancel challenge", "Cancel")}
                                    </button>
```


because then you get the content aligned with the continuation:

```
        <button className='danger sm' 
            onClick={this.cancelOwnChallenges.bind(self, this.state.rengo_list)}>
            {pgettext("Cancel challenge", "Cancel")}
       </button>
```
       
         
... wait, what, is that {pgettext part of the <button, or the content, ??

Reply:
Note: I think my ongoing attempts to reduce line length are improving readability, but at the moment getting consistently near 80chars is a stretch...

Reply:
What about...

```
                                    <button 
                                        className='danger sm' 
                                        onClick={this.cancelOwnChallenges.bind(self, this.state.rengo_list)}
                                    >
                                        {pgettext("Cancel challenge", "Cancel")}
                                    </button>
```

Reply:
Yeah - that's better than mine I think, but it still takes a human to work out that the > needs to go on the next line for readability

Reply:
If there was a rule saying something like `If the contents of any bracket ([{<>}]) covers multiple lines, indenting of contents and vertical alignment of bracket is appropriate? Feel like a robot could follow that fairly well, but I'm wondering if there is a situation we wouldn't want that to happen?

Reply:
Right, but in any case try telling prettifier that ... ?

Reply:
> it still takes a human to work out that the > needs to go on the next line for readability

> If there was a rule saying something like If the contents of any bracket ([{<>}]) covers multiple lines, indenting of contents and vertical alignment of the bracket is appropriate? Feel like a robot could follow that fairly well

> Right, but in any case try telling prettifier that ... ?

[If only there were a way to settle this...](https://prettier.io/docs/en/options.html#bracket-line)

Reply:
Lovely <3 I personally think the latter looks just a little bit nicer, but the 80pw is perfectly readable :)

Reply:
>I personally think the latter looks just a little bit nicer, but the 80pw is perfectly readable

I think I feel about the same.  And I will point out that with 80pw, no side scrolling necessary!

<details> 
  <summary>screenshot</summary>
  <img width="848" alt="Screen Shot 2022-01-05 at 12 44 30 PM" src="https://user-images.githubusercontent.com/25233703/148287145-4e458c0b-6086-453a-82c3-af9f4c7a1f6c.png">
</details>

Reply:
I am a huge fan of no side scrolling! ❤️

Reply:
Is this about side-scrolling?  Maybe you need a monitor upgrade? 

VSCode on my desktop has 2x 120 column windows _and_ the finder left pane open...

Reply:
He's talking about GitHub not VSCode

Reply:
Ah hah - yes indeed, this github layout is curiously narrow isn't it :D  :D